src/useImageDimensions.ts

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+/**
+ * @param source either a remote URL or a local file resource.
+ * @returns original image width and height.
+ */
+function useImageDimensions(source: ImageRequireSource | URISource) {
+  const [state, setState] = useState<{
+    width?: number
+    height?: number
+    loading?: boolean
+    error?: any
+  }>({})
```

Comment:
Can you provide initial values here, otherwise e.g. `loading` will be `undefined` in the first render loop, instead of `true`

Reply:
Now the hook returns loading property in proper boolean, not undefined